internal/controller/cluster_monitoring_controller.go

```diff
@@ -142,6 +151,12 @@ func (r *ClusterMonitoringReconciler) reconcileDelete(ctx context.Context, clust
 				logger.Error(err, "failed to delete heartbeat")
 				return ctrl.Result{RequeueAfter: 5 * time.Minute}, errors.WithStack(err)
 			}
+
+			err = r.MimirService.DeleteMimirSecrets(ctx)
```

Comment:
So now, we probably either should have CreateMimirSecrets and DeleteMimirSecrets or ConfigureMimir and UnconfigureMimir right? not a blocker though

Reply:
You're right. Removed it